docs/zkapps/tutorials/10-account-updates.mdx

```diff
@@ -0,0 +1,381 @@
+
+// creates a png file of a transaction, and saves it to a path
+saveTxn(txn: Mina.Transaction, name: string, legend: Legend, path: string)
+
+// prints a nicely formatted view of a transaction
+showTxn(txn: Mina.Transaction, name: string, legend: Legend)
+
+// with legend type, to replace public keys with human readable strings:
+type Legend = { [pk: string]: string };
+
+```
+
+We'll build our legend as follows:
+
+```typescript
+  const legend = {
+    [proofsOnlyAddr.toBase58()]: 'proofsOnlyZkApp',
+    [secondaryAddr.toBase58()]: 'secondaryZkApp',
+    [deployerAccount.toPublicKey().toBase58()]: 'deployer',
+  };
+```
+
+Then we'll deploy our smart contracts as follows, and visualize the transaction:
+
+```typescript
+  const deploy_txn = await Mina.transaction(deployerAccount, () => {
+    AccountUpdate.fundNewAccount(deployerAccount, 2);
+    proofsOnlyInstance.deploy({ zkappKey: proofsOnlySk });
+    secondaryInstance.deploy({ zkappKey: secondarySk });
+  });
+
+  await deploy_txn.prove();
+  deploy_txn.sign([ proofsOnlySk, secondarySk ]);
+
+  await showTxn(deploy_txn, 'deploy_txn', legend);
+
+  await deploy_txn.send();
+```
+
+This yields the following visualization of `deploy_txn`. 
+
+![](https://raw.githubusercontent.com/es92/zkApp-examples/main/10-account-updates/deploy_txn.png)
+
+This is best opened in a new tab to read the contents ([link](https://raw.githubusercontent.com/es92/zkApp-examples/main/10-account-updates/deploy_txn.png)).
+
+The deploy transaction includes 5 accountUpdates. In order from left to right;
+
+1. This accountUpdate takes the new account fee from the deployer for deploying the zkApps. Note the `-2` on the `balanceChange` field
+2. This update deploys the `proofsOnlyZkApp` instance. Note the permissions, all set to the values in the zkApp's deploy field, and the `preconditions`, asserting its a new account.
+3. This update initializes the `proofsOnlyZkApp`. Note the precondition that it can't already be in a proved state.
+4. This update deploys an instance of `secondaryZkApp`. Note the permissions here set to default compared to the deployment in the `proofsOnlyZkApp` example.
+5. This update initializes the `secondaryZkApp` instance.
+
+When the transaction is run on-chain, all of these accountUpdates will be checked, and applied. Note that there is no actual "execution" of each AccountUpdate provided - instead each AccountUpdate includes an associated signature or proof, corresponding to the verification key in the zkApp.
+
+Next, we'll call `add` on our instance of `proofsOnlyZkApp`:
+
+```typescript
+  const txn1 = await Mina.transaction(deployerAccount, () => {
+    proofsOnlyInstance.add(Field(4));
+  });
+
+  await txn1.prove();
+
+  await showTxn(txn1, 'txn1', legend);
+
+  await txn1.send();
+```
+
+This yields the following visualization of `txn1`
+
+![](https://raw.githubusercontent.com/es92/zkApp-examples/main/10-account-updates/txn1-copy.png)
```

Comment:
for all generated links, please apply the same feedback on including the username as part of the link

Reply:
So this was actually for just including a static link to images used in the tutorial - so should be my repo I think (so the images show up).

Probably should eventually be the docs repo, if there is a place images should be going (currently in `/examples/zkapps/10-account-updates`, which not sure reachable for rendering here)

Reply:
the convention to store images is [static/img](https://github.com/o1-labs/docs2/tree/main/static/img) 
<img width="359" alt="image" src="https://user-images.githubusercontent.com/10067215/230656240-7aa8bcbf-733c-4bfa-8ce0-f1ad045113d4.png">

Reply:
done, thanks